zebra-network/src/peer_set/inventory_registry.rs

```diff
@@ -0,0 +1,60 @@
+use crate::{protocol::external::InventoryHash, BoxedStdError};
```

Comment:
I tend to order imports from 'farthest to closest, top to bottom', so this would go under all the other imports as they are, I don't know if there's a Rust convention for that

Reply:
afaik there's no convention @dconnolly, I've always just bundled all includes into a single section and let rustfmt order them alphabetically. I don't love the splitup method because I always add `use` statements via `rust-analyzer`'s code actions. When we split up the use statements into multiple sections it always adds the new use statements to the last section. By keeping them together this works out because saving it makes it immediately reorder alphabetically thanks to rustfmt.

Reply:
There's a rustfmt option that we could enable that groups imports together, then we might have less to worry about.

Reply:
:o, I'd be down with that, so long as it's automatic.

Reply:
looks like they only have things for automatically merging / laying out the imports, not for separating them into groups.

https://rust-lang.github.io/rustfmt/?version=master&search=import

Reply:
trialed here: https://github.com/ZcashFoundation/zebra/pull/986/files

This merges all the imports that are possible to merge, but it doesn't merge e.g., `super::mod` with `crate::something`, which is IMO good because those imports should be used in different ways.

Reply:
makes sense, though I don't think the issue was whether or not use statements are merged. The question was should they be separated by newlines and sorted in sub groupings or should they all be placed into one group and ordered alphabetically.